iceoryx_hoofs/include/iceoryx_hoofs/internal/relocatable_pointer/pointer_repository.inl

```diff
@@ -71,11 +72,11 @@ inline id_t PointerRepository<id_t, ptr_t, CAPACITY>::registerPtr(const ptr_t pt
             {
                 m_maxRegistered = id;
             }
-            return id;
+            return cxx::make_optional<id_t>(id);
```

Comment:
I think it is not needed, we can just use 
```suggestion
            return id;
```
and the ctor will be invoked (as the compiler knows the return type). Unless it is explicit ... which is another reason I do not like explicit as this verbosity does not help here. While reading the impl code the only relevant information is that some id is returned (and not a null-value). We only care about the wrapper type at call site.